From 495f5184cdf2b9a38d875aa94f6280c9a9db8084 Mon Sep 17 00:00:00 2001
From: Arnaldo Garcia Rincon <agarrcia@amazon.com>
Date: Thu, 26 Mar 2026 18:29:59 +0000
Subject: [PATCH] Revert "kbuild: hide tools/ build targets from external
 module builds"

This reverts commit 1bb0b18a06dceee1fdc32161a72e28eab6f011c4 in which
the targets to build "tools/*" were hidden for external modules, but
they are required by the kmod kit since the 'tools/*' binaries are not
distributed as part of the archive.

Signed-off-by: Arnaldo Garcia Rincon <agarrcia@amazon.com>
[jepiyush: adjust context for 6.18.15]
Signed-off-by: Piyush Jena <jepiyush@amazon.com>
[jpculp: adjust context for 6.18.20]
Signed-off-by: Patrick J.P. Culp <jpculp@amazon.com>
---
 Makefile | 17 +++++++++--------
 1 file changed, 9 insertions(+), 8 deletions(-)

diff --git a/Makefile b/Makefile
index a4847a1..5ad4dd9 100644
--- a/Makefile
+++ b/Makefile
@@ -2,14 +2,6 @@ ifneq ($(wildcard $(objtool_O)),)
 	$(Q)$(MAKE) -sC $(abs_srctree)/tools/objtool O=$(objtool_O) srctree=$(abs_srctree) $(patsubst objtool_%,%,$@)
 endif
 
-tools/: FORCE
-	$(Q)mkdir -p $(objtree)/tools
-	$(Q)$(MAKE) O=$(abspath $(objtree)) subdir=tools -C $(srctree)/tools/
-
-tools/%: FORCE
-	$(Q)mkdir -p $(objtree)/tools
-	$(Q)$(MAKE) O=$(abspath $(objtree)) subdir=tools -C $(srctree)/tools/ $*
-
 # ---------------------------------------------------------------------------
 # Kernel selftest
 
@@ -19,6 +11,15 @@ kernelversion:
 image_name:
 	@echo $(KBUILD_IMAGE)
 
+tools/: FORCE
+	$(Q)mkdir -p $(objtree)/tools
+	$(Q)$(MAKE) O=$(abspath $(objtree)) subdir=tools -C $(srctree)/tools/
+
+tools/%: FORCE
+	$(Q)mkdir -p $(objtree)/tools
+	$(Q)$(MAKE) O=$(abspath $(objtree)) subdir=tools -C $(srctree)/tools/ $*
+
+
 PHONY += run-command
 run-command:
 	$(Q)$(KBUILD_RUN_COMMAND)
-- 
2.53.0

